### PR TITLE
feat: log send errors

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Implements a new error-handling strategy for better debugging.

Newsletter sending errors should now be stored in a meta as an array, collecting up to 10 most recent errors. If any, these errors should be displayed in a sidebar panel:

image

The most recent error, if the newsletter is not sent, should also be displayed as an error notification in the editor:
 
image

### How to test the changes in this Pull Request:

1. Check out this branch and configure your ESP to use Mailchimp
2. Add `return new WP_Error( 'some_error', 'Some error' );` to line `771` of `includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php`
3. Send a newsletter, and confirm you get the error immediate error: `Publishing failed. Some Error`
4. Refresh and confirm you get the persisted notification and the sidebar panel as shown above

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/newsletter-editor/debug-send/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		return null;
+	}
+	return (
+		<PluginDocumentSettingPanel
+			name="newsletters-ads-settings-panel"
+			title={ __( 'Debug Send Errors', 'newspack-newsletters' ) }
+		>
+			<p>
+				{ __(
+					'The following errors occurred when trying to send the newsletter:',
```
Comment (adekbadek) [inline src/newsletter-editor/debug-send/index.js:29]: Suggestion: mention that only last 10 errors are displayed.

File: src/newsletter-editor/debug-send/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			sent: meta.newsletter_sent,
+		};
+	} );
+	if ( sent || ! sendErrors?.length ) {
+		return null;
+	}
+	return (
+		<PluginDocumentSettingPanel
+			name="newsletters-ads-settings-panel"
+			title={ __( 'Debug Send Errors', 'newspack-newsletters' ) }
```
Comment (adekbadek) [inline src/newsletter-editor/debug-send/index.js:25]: Suggestion: 

```suggestion
			title={ __( 'Send Errors', 'newspack-newsletters' ) }
```

"debugging" feels too technical. But maybe it's just me.

**Review by adekbadek — APPROVED** [review]
Left some non-blocking suggestions.

File: src/newsletter-editor/debug-send/index.js (hunk 1 above)
Comment (miguelpeixe) [reply src/newsletter-editor/debug-send/index.js:29]: Added in da1f6c7b79ef3cf80d4901e3428b4f929786d937

**Review by adekbadek — APPROVED** [review]

**Pull request merged**